### ISSUE TITLE
classmethod for OrderBuilder to convert response from Client.get_order to OrderBuilder object

### ISSUE THREAD

**Issue by arainboldt** [alexgolec/tda-api#191]
**Describe the solution you'd like**

I've designing an application using the api that uses the OrderBuilder object as the unit for managing and manipulating orders within the application. 

As these orders, however, are stored in the database as json objects, i.e. returns from OrderBuilder.build and Client.get_order, I'll need to build a method for converting easily from order_spec to OrderBuilder. 

I'm currently going to build a helper func, but it seems like it might be something commonly encountered in which case the most appropriate implementation would be as a class method for OrderBuilder. 

perhaps something like: OrderBuilder.from_spec, OrderBuilder.from_json, etc.

tldr: A class method that would facilitate the conversion of order specs from the official api to OrderBuilder objects

**Describe alternatives you've considered**

An external helper func to parse the order spec and instantiate the OrderBuilder object

**Comment by arainboldt** [comment]
no guarantee of completeness here, or succintness for that matter, but the code I'm currently working with is:


```
from tda.orders import common as order_enums
from tda.orders.generic import OrderBuilder

order_field_to_enum_map = dict(

    orderType = 'OrderType',
    session = 'Session',
    duration = 'Duration',
    orderStrategyType = 'OrderStrategyType',
    orderLegCollection = 'OrderLegCollection',
    complexOrderStrategyType = 'ComplexOrderStrategyType',
    #childOrderStrategies = 'ChildOrderStrategies',

)

leg_field_to_enum_map = dict(

    OPTION = dict( 
    
            instruction = 'OptionInstruction',
            instrument = 'OptionInstrument',
            
    ),
    
    EQUITY = dict(
    
            instruction = 'EquityInstruction',
            instrument = 'EquityInstrument'
    
    ),


)

field_to_order_method_map = dict(

    orderType = 'order_type',
    session = 'session',
    duration = 'duration',
    orderStrategyType = 'order_strategy_type',
    orderLegCollection = None,
    complexOrderStrategyType = 'complex_order_strategy_type',
    price = 'price',
    quantity = 'quantity',
    instruction = 'instruction',
    OPTION = 'add_option_leg',
    EQUITY = 'add_equity_leg',

)



def create_instrument(instrument_spec):
    
    assetType = instrument_spec['assetType']
    
    symbol = instrument_spec['symbol']
    
    return getattr(order_enums,leg_field_to_enum_map[assetType]['instrument'])(symbol)
    

def extract_leg_kwargs(leg_spec):
    
    assetType = leg_spec['instrument']['assetType']

    instruction_enum = leg_field_to_enum_map[assetType]['instruction']
    
    kwargs = dict(
    
        instruction = getattr( 
                                getattr(
                                    order_enums,
                                    instruction_enum,
                                ),
                            leg_spec['instruction']
        ),

        symbol = leg_spec['instrument']['symbol'],

        quantity = leg_spec['quantity'],
        
    )
    
    return kwargs
    
def get_order_enum(field,val):
    
    enum_name = order_field_to_enum_map.get(field,None)
    
    if enum_name:
        
        enum = getattr( 
                    getattr(order_enums, enum_name),
                    val
        )
        
    else:
        
        enum = val
        
    return enum
    

def order_from_spec(order_spec):
    
    order = OrderBuilder(enforce_enums=True)
    
    for field, val in order_spec.items():
        
        if field == 'orderLegCollection':
            
            for leg_spec in val:
            
                leg_kwargs = extract_leg_kwargs(leg_spec)
                
                add_leg_method_name = field_to_order_method_map[leg_spec['instrument']['assetType']]
                
                getattr(order,add_leg_method_name)(**leg_kwargs)
                
            continue
            
        method_name_part = field_to_order_method_map.get(field,None)
        
        if method_name_part:
        
            set_method_name = f'set_{method_name_part}'
            
            enum = get_order_enum(field,val)

            getattr(order,set_method_name)(enum)
        
    return order

```

Then given the above, the below should be true for an arbitrary order.

```
call_spread = {
  "orderType": "NET_DEBIT",
  "session": "NORMAL",
  "price": "1.20",
  "duration": "DAY",
  "orderStrategyType": "SINGLE",
  "orderLegCollection": [
    {
      "instruction": "BUY_TO_OPEN",
      "quantity": 10,
      "instrument": {
        "symbol": "XYZ_011516C40",
        "assetType": "OPTION"
      }
    },
    {
      "instruction": "SELL_TO_OPEN",
      "quantity": 10,
      "instrument": {
        "symbol": "XYZ_011516C42.5",
        "assetType": "OPTION"
      }
    }
  ]
}

order_from_spec(call_spread).build() == call_spread

```

**Comment by alexgolec** [comment]
Ooh this is a very interesting proposal. Lately I've been seeing a lot of requests for help with creating these sorts of orders, and I'd like to take this on myself, complete with adding documentation.
